File: SpearoGo/Views/ConditionsPage.swift
import SwiftUI

struct ConditionsPage: View {
    @Environment(AppState.self) private var appState

    var body: some View {
        ZStack {
            Color(hex: Constants.Colors.background).ignoresSafeArea()

            VStack(spacing: 10) {
                Text("CONDITIONS")
                    .font(.system(size: 10, weight: .semibold))
                    .foregroundStyle(Color(hex: Constants.Colors.textSecondary))
                    .kerning(1.5)

                if let weather = appState.weatherData, let marine = appState.marineData {
                    HStack(spacing: 20) {
                        ConditionItem(
                            icon: "wind",
                            label: "WIND",
                            value: String(format: "%.0f", weather.windSpeed),
                            unit: "kn"
                        )
                        ConditionItem(
                            icon: "water.waves",
                            label: "SWELL",
                            value: String(format: "%.1f", marine.waveHeight),
                            unit: "m"
                        )
                    }

                    HStack(spacing: 20) {
                        ConditionItem(
                            icon: "arrow.up.right",
                            label: "DIRECTION",
                            value: compassDirection(degrees: weather.windDirection),
                            unit: ""
                        )
                        ConditionItem(
                            icon: "timer",
                            label: "PERIOD",
                            value: String(format: "%.0f", marine.wavePeriod),
                            unit: "s"
                        )
                    }
                } else {
                    ProgressView()
                        .tint(Color(hex: Constants.Colors.primaryAccent))
                }
            }
            .padding()
        }
    }

    private func compassDirection(degrees: Double) -> String {
        let directions = ["N","NE","E","SE","S","SW","W","NW"]
        let index = Int((degrees + 22.5) / 45.0) % 8
        return directions[index]
    }
}

struct ConditionItem: View {
    let icon: String
    let label: String
    let value: String
    let unit: String

    var body: some View {
        VStack(spacing: 2) {
            Image(systemName: icon)
                .font(.system(size: 14))
                .foregroundStyle(Color(hex: Constants.Colors.primaryAccent))
            Text(label)
                .font(.system(size: 8, weight: .medium))
                .foregroundStyle(Color(hex: Constants.Colors.textSecondary))
                .kerning(0.8)
            HStack(alignment: .lastTextBaseline, spacing: 1) {
                Text(value)
                    .font(.system(size: 18, weight: .bold))
                    .foregroundStyle(Color(hex: Constants.Colors.textPrimary))
                if !unit.isEmpty {
                    Text(unit)
                        .font(.system(size: 9))
                        .foregroundStyle(Color(hex: Constants.Colors.textSecondary))
                }
            }
        }
        .frame(minWidth: 55)
    }
}

File: wear/app/src/main/java/com/spearotracker/spearogo/ui/pages/ConditionsPage.kt
package com.spearotracker.spearogo.ui.pages

import androidx.compose.foundation.layout.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.wear.compose.material3.Text
import com.spearotracker.spearogo.ui.AppUiState
import com.spearotracker.spearogo.ui.components.ConditionItem
import com.spearotracker.spearogo.ui.components.ConditionRow
import com.spearotracker.spearogo.ui.components.ScrollingPage
import com.spearotracker.spearogo.ui.components.ConditionItemSkeleton
import com.spearotracker.spearogo.ui.theme.Brand
import androidx.compose.ui.text.style.TextAlign

@Composable
fun ConditionsPage(uiState: AppUiState) {
    ScrollingPage {
        Text(
            text = "CONDITIONS",
            style = Brand.Typography.sectionHeader,
            color = Brand.Colors.textSecondary,
            modifier = Modifier.padding(bottom = Brand.Spacing.item)
        )

        val weather = uiState.weatherData
        val marine = uiState.marineData

        // Wind is shown whenever weather is known, even where there is no
        // sea. Requiring marine data here hid the wind at any location the
        // marine API does not cover.
        if (weather != null) {
            ConditionRow(modifier = Modifier.padding(bottom = Brand.Spacing.section)) {
                ConditionItem(icon = "wind", label = "Wind", value = "%.0f".format(weather.windSpeed), unit = "kn")
                ConditionItem(
                    icon = "waves",
                    label = "Swell",
                    value = marine?.let { "%.1f".format(it.waveHeight) } ?: "—",
                    unit = if (marine == null) "" else "m"
                )
            }
            ConditionRow {
                ConditionItem(icon = "direction", label = "Dir", value = compassDirection(weather.windDirection), unit = "")
                ConditionItem(
                    icon = "timer",
                    label = "Period",
                    value = marine?.wavePeriod?.let { "%.0f".format(it) } ?: "—",
                    unit = if (marine?.wavePeriod == null) "" else "s"
                )
            }
            if (marine == null) {
                Text(
                    text = "No swell data for this spot",
                    style = Brand.Typography.caption,
                    color = Brand.Colors.textSecondary,
                    textAlign = TextAlign.Center,
                    modifier = Modifier.padding(top = Brand.Spacing.item)
                )
            }
        } else {
            ConditionRow(modifier = Modifier.padding(bottom = Brand.Spacing.section)) {
                ConditionItemSkeleton()
                ConditionItemSkeleton()
            }
            ConditionRow {
                ConditionItemSkeleton()
                ConditionItemSkeleton()
            }
        }
    }
}

fun compassDirection(degrees: Double): String {
    val dirs = arrayOf("N", "NE", "E", "SE", "S", "SW", "W", "NW")
    return dirs[((degrees + 22.5) / 45.0).toInt() % 8]
}
